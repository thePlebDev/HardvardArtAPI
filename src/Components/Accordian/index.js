import React,{useState} from 'react';
import styled from 'styled-components'

import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';

const Container = styled.div`
  border:1px solid red;
  display:flex;
  flex-direction:column;
  align-items:center;
  transition:all .25s;
  height:${({state})=>state ? "100px" : '20px'};
  overflow:hidden;


`
const Arrow = styled.span`

  transition:all .25s;
  transform:${({state})=>state ? 'rotate(180deg)': 'rotate(0deg)'};
`
const ArrowContainer = styled.div`
  display:flex;
  justify-content:center;
  align-items:center;
`


const Accordian =()=>{
  const [state,setState] = useState(false)

  return(
    <Container state={state}>
    <ArrowContainer onClick={()=>setState(!state)}>
      <span>Advanced Search</span>
      <Arrow state={state}>
        <KeyboardArrowUpIcon/>
      </Arrow>
      </ArrowContainer>

      <div>Period</div>
      <div>Place</div>
      <div>Century</div>
    </Container>
  )
}

export default Accordian
