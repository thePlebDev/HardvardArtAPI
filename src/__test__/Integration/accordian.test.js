import React from 'react';
import Accordian from '../../Components/Accordian'

import {render, screen} from '@testing-library/react'


describe('testing the Accordian component',()=>{
  it('should test the snapshot',()=>{
    const {asFragment} = render(<Accordian/>)
    expect(asFragment()).toMatchSnapshot();

  })

  it('should render all the important stuff',()=>{
    const {getByTestId,getByText} = render(<Accordian/>)
    let arrow = getByTestId('arrow')
    
  })
})
