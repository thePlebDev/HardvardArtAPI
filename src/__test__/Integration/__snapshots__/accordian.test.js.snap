// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Accordian component should test the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnbm iIfCDR"
    height="0"
  >
    <div>
      <div
        class="sc-eCssSg gThwce"
      >
        <span>
          Advanced Search
        </span>
        <span
          class="sc-hKgILt cDnqYS"
          data-testid="arrow"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
            />
          </svg>
        </span>
      </div>
      <div
        class="sc-bdfBwQ kDdlMT"
      >
        <div
          style="display: flex; align-items: center; justify-content: center;"
        >
          Person
          <span
            class="sc-gsTCUz jGiHyD"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="sc-bdfBwQ kDdlMT"
      >
        <div
          style="display: flex; align-items: center; justify-content: center;"
        >
          Places
          <span
            class="sc-gsTCUz jGiHyD"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="sc-bdfBwQ kDdlMT"
      >
        <div
          style="display: flex; align-items: center; justify-content: center;"
        >
          Period
          <span
            class="sc-gsTCUz jGiHyD"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
